rkt/containers.go

```diff
@@ -0,0 +1,178 @@
+// Copyright 2014 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/coreos/rocket/pkg/lock"
+)
+
+type containerHandle struct {
+	uuid       string
+	lock       *lock.DirLock
+	isExited   bool
+	isGarbage  bool
+	isDeleting bool
+}
+
+type includeMask byte
+
+const (
+	includeContainersDir includeMask = 1 << iota
+	includeGarbageDir
+)
+
+// withContainerHandles iterates over the included containers calling function f for every container.
+func withContainerHandles(include includeMask, f func(*containerHandle) error) error {
```

Comment:
IMO `walkContainers` (or `mapContainers`) would be fine; I find the `with` slightly misleading because it implies a _collective_ context rather than per-container